test/unit/gateways/adyen_test.rb

```diff
@@ -1332,6 +1332,80 @@ def test_level_3_data
     assert_success response
   end
 
+  def test_succesful_additional_airline_data
+    airline_data = {
+      agency_invoice_number: 'BAC123',
+      agency_plan_name: 'plan name',
+      airline_code: '434234',
+      airline_designator_code: '1234',
+      boarding_fee: '100',
+      computerized_reservation_system: 'abcd',
+      customer_reference_number: 'asdf1234',
+      document_type: 'cc',
+      leg: {
+        carrier_code: 'KL'
+      },
+      passenger: {
+        first_name: 'Joe',
+        last_name: 'Doe'
+      }
+    }
+
+    response = stub_comms do
+      @gateway.authorize(@amount, @credit_card, @options.merge(additional_data_airline: airline_data))
+    end.check_request do |_endpoint, data, _headers|
+      parsed = JSON.parse(data)
+      additional_data = parsed['additionalData']
+      assert_equal additional_data['airline.agency_invoice_number'], airline_data[:agency_invoice_number]
```

Comment:
💬 Comment:

I know that is a little bit annoying but this test is the only checkpoint that verifies that airline data mapping is correct (gateway is not going to tell you if something is missing), so I think Is a good idea to include all the fields.